FIAT/reference_element.py

```diff
@@ -623,6 +623,25 @@ def contains_point(self, point, epsilon=0):
             result &= (c - epsilon <= 1)
         return result
 
+    def factor_entity(self, entity):
```

Comment:
[β-reduction](https://en.wikipedia.org/wiki/Lambda_calculus#Beta_reduction)

We introduced `factor_entity` so that we can dispatch on the cell to decide how to do something. I think it is now clear which implementation will `factor_entity` resolve to at each place where it is called.